app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java

```diff
@@ -184,161 +181,217 @@ public boolean contactExists(final String name) throws IOException {
         return getContactWithName(name) != null;
     }
 
+
     /**
      * {@inheritDoc}
      */
     @Override
-    public List<Profile> getProfiles() throws IOException {
-        final GsonBuilder gsonBuilder = new GsonBuilder()
-                .registerTypeHierarchyAdapter(IKey.class, new GsonKeyAdapter());
-        final Gson gsonParser = gsonBuilder.create();
-
-        final Reader reader = this.fileSystem.getReader(this.profilesPath);
-        final List<Profile> profile = gsonParser.fromJson(reader, TYPE_PROFILE_LIST);
-        reader.close();
-
-        return profile;
+    public Profile getProfile() throws IOException {
+        return getDatabase().getProfile();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public void addProfile(final Profile profile) throws IOException {
-        // Get the list of profiles and add the new profile
-        final List<Profile> profiles = this.getProfiles();
-        profiles.add(profile);
-
-        final GsonBuilder gsonBuilder = new GsonBuilder()
-                .registerTypeHierarchyAdapter(IKey.class, new GsonKeyAdapter());
-        final Gson gsonParser = gsonBuilder.create();
+    public void updateProfile(final Profile newProfile) throws IOException {
+        getDatabase().setProfile(newProfile);
```

Comment:
This getter is used to ensure that there is a database present in the databasemap, or in other words that the database has been created and loaded